pystac/stac_object.py

```diff
@@ -507,6 +508,11 @@ def from_dict(
                 previously resolved instances of the STAC object.
             migrate: Use True if this dict represents JSON from an older STAC object,
                 so that migrations are run against it.
+            preserve_dict: If False, the dict parameter ``d`` may be modified
+                during this method call. Otherwise the dict is not mutated.
+                Defaults to True, which results results in a deepcopy of the
+                parameter. Set to False when possible to avoid the performance
+                hit of a deepcopy. Defaults to True.
```

Comment:
```suggestion
                hit of a deepcopy.
```

Already state earlier